### PR TITLE
FOIA-237: Inline documentation and README for foia_ui module.

### PR DESCRIPTION
**Pull request by amanire**
- Add `README.md`.
- Fix typos and update docblock.
- Minor code cleanup.

### CODE REVIEW

File: docroot/modules/custom/foia_ui/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# FOIA UI CUSTOM MODULE
+
+## CONTENTS OF THIS FILE
+
+- Introduction
+- Requirements
+- Installation
+- Documentation
+
+## INTRODUCTION
```
Comment (bemarlan) [inline docroot/modules/custom/foia_ui/README.md]: Need two empty lines prior to headings (except first one).

File: docroot/modules/custom/foia_ui/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+The FOIA UI module contains custom user interface modifications for the DOJ FOIA
+project. This includes the following:
+
+- OpenAPI Docs generation and configuration
```
Comment (bemarlan) [inline docroot/modules/custom/foia_ui/README.md]: For bullets there should be a space then an asterisk.

[space][asterisk][space][text]

File: docroot/modules/custom/foia_ui/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+```
+
+The jQuery selector for all instances of the field is:
+
+```js
+$("input[name*='field_foia_requests_va']").filter("input[name*='field_req_pend_start_yr'])
+```
+Where:
+
+- `field_foia_requests_va` is the paragraph type machine name.
```
Comment (bemarlan) [inline docroot/modules/custom/foia_ui/README.md]: For bullets there should be a space then an asterisk.

[space][asterisk][space][text]

File: docroot/modules/custom/foia_ui/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+### Utility JavaScript functions
+
+Several JavaScript text-processing functions are used in common between
+`foia_ui` validation and `foia_advcalc` auto-calculation. They can be accessed
+as methods on the `Drupal.FoiaUI` object, e.g.
+
+```js
+Drupal.FoiaUI.specialNumber("<1");
+// returns 0.1
+````
```
Comment (bemarlan) [inline docroot/modules/custom/foia_ui/README.md]: only 3 ` needed here

File: docroot/modules/custom/foia_ui/README.md (hunk 2 above)
Comment (amanire) [reply docroot/modules/custom/foia_ui/README.md]: You sure? My linter says the opposite:
> MD006/ul-start-left: Consider starting bulleted lists at the beginning of the line [Expected: 0; Actual: 1]markdownlint(MD006)

File: docroot/modules/custom/foia_ui/README.md (hunk 1 above)
Comment (amanire) [reply docroot/modules/custom/foia_ui/README.md]: I don't really care either way, but my markdown linter has a rule against that.
> MD012/no-multiple-blanks: Multiple consecutive blank lines [Expected: 1; Actual: 2]markdownlint(MD012)

File: docroot/modules/custom/foia_ui/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+Several JavaScript text-processing functions are used in common between
+`foia_ui` validation and `foia_advcalc` auto-calculation. They can be accessed
+as methods on the `Drupal.FoiaUI` object, e.g.
+
+```js
+Drupal.FoiaUI.specialNumber("<1");
+// returns 0.1
+```
+
```
Comment (bemarlan) [inline docroot/modules/custom/foia_ui/README.md]: Can you add another line break here?

File: docroot/modules/custom/foia_ui/README.md (hunk 3 above)
Comment (bemarlan) [reply docroot/modules/custom/foia_ui/README.md]: Change bullets

Comment (amanire) [reply docroot/modules/custom/foia_ui/README.md]: Sorry, I missed these comments earlier!

File: docroot/modules/custom/foia_ui/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+module. Visit
+https://www.drupal.org/docs/8/extending-drupal-8/installing-drupal-8-modules#s-step-2-enable-the-module
+for further information.
+
+
+DOCUMENTATION
+-------------
+
+
+### Open API Docs
```
Comment (bemarlan) [inline docroot/modules/custom/foia_ui/README.md]: Final small change--can we add these h3s to the ToC nested under Documentation?

Comment (jesseday) [reply docroot/modules/custom/foia_ui/README.md]: Done

**Pull request merged**